### PR TITLE
[Logs]: Allow logs-agent to send logs to a different account

### PR DESCRIPTION
**Pull request by gaetan-deputier**
### What does this PR do?

Allow logs-agent to send logs to a different account by providing another api key under logs_config.logs_api_key.

### Motivation

We have a use case where some teams wants send their metrics to US and logs to EU. One solution is to allow the logs-agent to have a secondary Api key.

### CODE REVIEW

**Review by mnshdw — DISMISSED** [review]

File: pkg/logs/sender/config.go [hunk 1]
```diff
@@ -89,3 +84,15 @@ func BuildEndpoints() (*client.Endpoints, error) {
...
 	return client.NewEndpoints(main, additionals), nil
 }
+
+func isSetAndNotEmpty(config config.Config, key string) bool {
+	return config.IsSet(key) && len(config.GetString(key)) > 0
+}
+
+//getLogsApiKey provides the dd api key used by the main logs agent sender.
+func getLogsAPIKey(config config.Config) string {
+	if isSetAndNotEmpty(config, "logs_config.logs_api_key") {
```
Comment (ajacquemot) [inline pkg/logs/sender/config.go]: do we have to add the `logs_` prefix ? we are already in the `logs_config` namespace

File: releasenotes/notes/releasenotes/notes/logs-override-api-key-5bbaa007fc80e975.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    Allow logs-agent to send logs to a different account by providing another api key under logs_config.logs_api_key.
```
Comment (ajacquemot) [inline releasenotes/notes/releasenotes/notes/logs-override-api-key-5bbaa007fc80e975.yaml]: I would definitively not advertise this

File: pkg/logs/sender/config.go (hunk 1 above)
Comment (gaetan-deputier) [reply pkg/logs/sender/config.go]: Right, I wanted to emphasise on the difference between this api key and the logs one but I agree, the namespace is enough.

Comment (gaetan-deputier) [reply pkg/logs/sender/config.go]: Updated

File: releasenotes/notes/releasenotes/notes/logs-override-api-key-5bbaa007fc80e975.yaml (hunk 2 above)
Comment (gaetan-deputier) [reply releasenotes/notes/releasenotes/notes/logs-override-api-key-5bbaa007fc80e975.yaml]: Updated

**Review by achntrl — DISMISSED** [review]
LGTM, we might want to bind it to the environment, in that case there one more line to add like this: 
https://github.com/DataDog/datadog-agent/commit/23bdfe36c0a60bba012fad7af6f38d1100228598#diff-091ace02b49a46f55f2586cfec89a160R299

**Comment by ajacquemot** [comment]
@achntrl I think this is something we don't want to do as we want to limit the use of this feature

**Comment by ajacquemot** [comment]
@gaetan-deputier did you consider adding a main endpoint configuration that would look the same as the additional endpoints configuration so that it unifies things ?

**Comment by gaetan-deputier** [comment]
@ajacquemot Do you mean with a set of attributes (apikey/host/port)? I did not want to introduce another set of main attributes (There are logs_dd_url and dd_url + dd_port already for host:port).
Now, overall I tend to agree to have a unified "host" config definition with a triplet (key, host, port) for both main and additional would be better

**Comment by ajacquemot** [comment]
@gaetan-deputier there would be just one main attribute, I just wanted to share this thought with you, this is not a blocker, we are just trying not to leak to many configuration parameters because we know we can't delete them afterward, otherwise it could break on the customer side.

**Review by ajacquemot — DISMISSED** [review]

File: pkg/logs/sender/config.go [hunk 3]
```diff
@@ -89,3 +84,15 @@ func BuildEndpoints() (*client.Endpoints, error) {
...
+		return config.GetString("logs_config.logs_api_key")
```
Comment (arbll) [inline pkg/logs/sender/config.go]:
```suggestion
		return config.GetString("logs_config.api_key")
```

File: pkg/logs/sender/config.go (hunk 1 above)
Comment (arbll) [inline pkg/logs/sender/config.go]: I think you forgot to update `logs_api_key` to `api_key` here and bellow.
```suggestion
	if isSetAndNotEmpty(config, "logs_config.api_key") {
```

File: pkg/logs/sender/config_test.go [hunk 4]
```diff
@@ -145,6 +145,23 @@ func (suite *ConfigTestSuite) TestIsSetAndNotEmpty() {
...
+	suite.config.Set("api_key", "wassupkey")
+	suite.Equal("wassupkey", getLogsAPIKey(suite.config))
+	endpoints, err := BuildEndpoints()
+	suite.Nil(err)
+	suite.Equal("wassupkey", endpoints.Main.APIKey)
+}
+
+func (suite *ConfigTestSuite) TestOverrideApiKey() {
+	suite.config.Set("api_key", "wassupkey")
+	suite.config.Set("logs_config.logs_api_key", "wassuplogskey")
```
Comment (arbll) [inline pkg/logs/sender/config_test.go]:
```suggestion
	suite.config.Set("logs_config.api_key", "wassuplogskey")
```

**Review by arbll — CHANGES_REQUESTED** [review]

**Review by arbll — APPROVED** [review]
📦

**Pull request merged**